### ISSUE TITLE
Lotus worker process C2 process ran out of memory

### ISSUE THREAD

**Issue by William8Work** [filecoin-project/lotus#7528]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [X] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
$ lotus version && lotus-miner version
Daemon:  1.13.0-rc2+mainnet+git.530011fd0+api1.4.0
Local: lotus version 1.13.0-rc2+mainnet+git.530011fd0
Daemon:  1.13.0-rc2+mainnet+git.530011fd0+api1.2.0
Local: lotus-miner version 1.13.0-rc2+mainnet+git.530011fd0
```

### Describe the Bug

Since running this version of lotus worker, the C2 worker process has ran into memory problem several times in two different machines. Eventually the process took too much memory and caused the machine unresponsive.

However, not every sector runs into the same problem. It seems only after it successfully sealed multiple sectors, the process started to take a lot of memory and caused it got killed. 

Notes: this machine has been running C2 process for months. I have only seen this when running the cuda setting has been causing this out of memory problem. If recompiled lotus-worker without the cuda setting, the C2 process is working fine again. Maybe it is related to cuda setting?

`$ journalctl -n 100000 -u lotusworker2.service | grep -ai -A 10 -B10 'code=killed' `
```
Oct 11 22:46:35 g1wrk01 lotus-worker[115772]: 2021-10-11T22:46:35.874 INFO filcrypto::proofs::api > seal_commit_phase2: finish
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.690 INFO filcrypto::proofs::api > seal_commit_phase2: start
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.847 INFO filecoin_proofs::api::seal > seal_commit_phase2:start: SectorId(9911)
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.847 INFO filecoin_proofs::caches > trying parameters memory cache for: STACKED[34359738368]
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.847 INFO filecoin_proofs::caches > found params in memory cache for STACKED[34359738368]
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.847 INFO filecoin_proofs::api::seal > got groth params (34359738368) while sealing
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.847 INFO filecoin_proofs::api::seal > snark_proof:start
Oct 11 22:57:04 g1wrk01 lotus-worker[115772]: 2021-10-11T22:57:04.858 INFO bellperson::groth16::prover > Bellperson 0.17.0 is being used!
Oct 11 22:58:33 g1wrk01 lotus-worker[115772]: 2021-10-11T22:58:33.326 INFO bellperson::groth16::prover > synthesis time: 88.457675501s
Oct 11 22:58:33 g1wrk01 lotus-worker[115772]: 2021-10-11T22:58:33.328 INFO bellperson::groth16::prover > starting proof timer
Oct 12 00:11:18 g1wrk01 systemd[1]: lotusworker2.service: Main process exited, code=killed, status=9/KILL
Oct 12 00:11:18 g1wrk01 systemd[1]: lotusworker2.service: Failed with result 'signal'.
Oct 12 07:58:47 g1wrk01 systemd[1]: Started g1wrk012.
Oct 12 07:58:47 g1wrk01 lotus-worker[122471]: ignoring unrecognized log format 'console'
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.473 INFO filcrypto::proofs::api > seal_commit_phase2: start
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.632 INFO filecoin_proofs::api::seal > seal_commit_phase2:start: SectorId(9911)
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.633 INFO filecoin_proofs::caches > trying parameters memory cache for: STACKED[34359738368]
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.633 INFO filecoin_proofs::caches > no params in memory cache for STACKED[34359738368]
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.635 INFO storage_proofs_core::parameter_cache > parameter set identifier for cache: layered_drgporep::PublicParams{ graph: stacked_graph::StackedGraph{expansion_degree: 8 base_graph: drgraph::BucketGraph{size: 1073741824; degree: 6; hasher: poseidon_hasher} }, challenges: LayerChallenges { layers: 11, max_count: 18 }, tree: merkletree-poseidon_hasher-8-8-0 }
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.638 INFO storage_proofs_core::parameter_cache > ensuring that all ancestor directories for: "/media/cache/parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params" exist
Oct 12 08:01:50 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:50.638 INFO storage_proofs_core::parameter_cache > checking cache_path: "/media/cache/parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params" for parameters
Oct 12 08:01:52 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:52.930 INFO storage_proofs_core::parameter_cache > read parameters from cache "/media/cache/parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params"
Oct 12 08:01:52 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:52.930 INFO filecoin_proofs::api::seal > got groth params (34359738368) while sealing
Oct 12 08:01:52 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:52.931 INFO filecoin_proofs::api::seal > snark_proof:start
Oct 12 08:01:52 g1wrk01 lotus-worker[122471]: 2021-10-12T08:01:52.952 INFO bellperson::groth16::prover > Bellperson 0.17.0 is being used!
Oct 12 08:03:10 g1wrk01 lotus-worker[122471]: 2021-10-12T08:03:10.198 INFO bellperson::groth16::prover > synthesis time: 77.235612429s
Oct 12 08:03:10 g1wrk01 lotus-worker[122471]: 2021-10-12T08:03:10.198 INFO bellperson::groth16::prover > starting proof timer
Oct 12 09:52:31 g1wrk01 systemd[1]: lotusworker2.service: Main process exited, code=killed, status=9/KILL
Oct 12 09:52:31 g1wrk01 systemd[1]: lotusworker2.service: Failed with result 'signal'.
Oct 12 12:05:24 g1wrk01 systemd[1]: Started g1wrk012.
Oct 12 12:05:24 g1wrk01 lotus-worker[144657]: ignoring unrecognized log format 'console'
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.129 INFO filcrypto::proofs::api > seal_commit_phase2: start
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.300 INFO filecoin_proofs::api::seal > seal_commit_phase2:start: SectorId(9911)
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.300 INFO filecoin_proofs::caches > trying parameters memory cache for: STACKED[34359738368]
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.300 INFO filecoin_proofs::caches > no params in memory cache for STACKED[34359738368]
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.300 INFO storage_proofs_core::parameter_cache > parameter set identifier for cache: layered_drgporep::PublicParams{ graph: stacked_graph::StackedGraph{expansion_degree: 8 base_graph: drgraph::BucketGraph{size: 1073741824; degree: 6; hasher: poseidon_hasher} }, challenges: LayerChallenges { layers: 11, max_count: 18 }, tree: merkletree-poseidon_hasher-8-8-0 }
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.300 INFO storage_proofs_core::parameter_cache > ensuring that all ancestor directories for: "/media/cache/parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params" exist
Oct 12 12:08:46 g1wrk01 lotus-worker[144657]: 2021-10-12T12:08:46.300 INFO storage_proofs_core::parameter_cache > checking cache_path: "/media/cache/parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params" for parameters
--
Oct 16 08:41:19 g1wrk01 lotus-worker[193561]: 2021-10-16T08:41:19.389 INFO filcrypto::proofs::api > seal_commit_phase2: finish
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.080 INFO filcrypto::proofs::api > seal_commit_phase2: start
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.235 INFO filecoin_proofs::api::seal > seal_commit_phase2:start: SectorId(10080)
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.235 INFO filecoin_proofs::caches > trying parameters memory cache for: STACKED[34359738368]
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.235 INFO filecoin_proofs::caches > found params in memory cache for STACKED[34359738368]
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.235 INFO filecoin_proofs::api::seal > got groth params (34359738368) while sealing
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.235 INFO filecoin_proofs::api::seal > snark_proof:start
Oct 16 09:44:32 g1wrk01 lotus-worker[193561]: 2021-10-16T09:44:32.256 INFO bellperson::groth16::prover > Bellperson 0.17.0 is being used!
Oct 16 09:45:52 g1wrk01 lotus-worker[193561]: 2021-10-16T09:45:52.635 INFO bellperson::groth16::prover > synthesis time: 80.368779587s
Oct 16 09:45:52 g1wrk01 lotus-worker[193561]: 2021-10-16T09:45:52.636 INFO bellperson::groth16::prover > starting proof timer
Oct 16 11:19:12 g1wrk01 systemd[1]: lotusworker2.service: Main process exited, code=killed, status=9/KILL
Oct 16 11:19:12 g1wrk01 systemd[1]: lotusworker2.service: Failed with result 'signal'.
Oct 16 17:28:31 g1wrk01 systemd[1]: Started g1wrk012.
Oct 16 17:28:31 g1wrk01 lotus-worker[199984]: ignoring unrecognized log format 'console'
```

### Logging Information

```text
The system log:

Notes: there are three lotus-worker running in this single machine. two of them are PC2 which is fine. the other one is C2 process which caused memory problem.

$dmesg --time-format=iso

2021-10-16T11:17:37,027005-07:00 Tasks state (memory values in pages):
2021-10-16T11:17:37,027006-07:00 [  pid  ]   uid  tgid total_vm      rss pgtables_bytes swapents oom_score_adj name
2021-10-16T11:17:37,027011-07:00 [    681]     0   681   130787      400  1060864        0          -250 systemd-journal
2021-10-16T11:17:37,027014-07:00 [   1148]   128  1148     1778      108    53248        0             0 rpcbind
2021-10-16T11:17:37,027015-07:00 [   1170]     0  1170    59828      539   106496        0             0 accounts-daemon
2021-10-16T11:17:37,027016-07:00 [   1171]     0  1171      637       44    40960        0             0 acpid
2021-10-16T11:17:37,027017-07:00 [   1174]   115  1174     2121      103    57344        0             0 avahi-daemon
2021-10-16T11:17:37,027018-07:00 [   1175]     0  1175     2355       66    53248        0             0 cron
2021-10-16T11:17:37,027019-07:00 [   1177]   103  1177     2239      525    57344        0          -900 dbus-daemon
2021-10-16T11:17:37,027021-07:00 [   1185]     0  1185    20556      182    57344        0             0 irqbalance
2021-10-16T11:17:37,027022-07:00 [   1187]     0  1187     9832     2064   110592        0             0 networkd-dispat
2021-10-16T11:17:37,027023-07:00 [   1191]     0  1191    58755      981    90112        0             0 polkitd
2021-10-16T11:17:37,027024-07:00 [   1193]   104  1193    56089      518    81920        0             0 rsyslogd
2021-10-16T11:17:37,027025-07:00 [   1195]     0  1195    58900      141    86016        0             0 switcheroo-cont
2021-10-16T11:17:37,027026-07:00 [   1202]     0  1202    98278      801   122880        0             0 udisksd
2021-10-16T11:17:37,027027-07:00 [   1204]   127  1204     1302       41    57344        0             0 nvidia-persiste
2021-10-16T11:17:37,027028-07:00 [   1207]     0  1207     3422      160    65536        0             0 wpa_supplicant
2021-10-16T11:17:37,027029-07:00 [   1228]   115  1228     2088       82    57344        0             0 avahi-daemon
2021-10-16T11:17:37,027030-07:00 [   1298]     0  1298    78439      457   114688        0             0 ModemManager
2021-10-16T11:17:37,027031-07:00 [   1314]     0  1314     3046      230    65536        0         -1000 sshd
2021-10-16T11:17:37,027032-07:00 [   1315]     0  1315    59902      292    98304        0             0 gdm3
2021-10-16T11:17:37,027033-07:00 [   1338]     0  1338    29467     2059   126976        0             0 unattended-upgr
2021-10-16T11:17:37,027035-07:00 [   1386]   111  1386    38235       74    57344        0             0 rtkit-daemon
2021-10-16T11:17:37,027036-07:00 [   1489]     0  1489    63092      396   106496        0             0 upowerd
2021-10-16T11:17:37,027037-07:00 [   1776]   121  1776    61584     1315   114688        0             0 colord
2021-10-16T11:17:37,027038-07:00 [   1841]   120  1841    81704      536   139264        0             0 whoopsie
2021-10-16T11:17:37,027039-07:00 [   1844]   116  1844     2816      119    65536        0             0 kerneloops
2021-10-16T11:17:37,027040-07:00 [   1846]   116  1846     2816      119    65536        0             0 kerneloops
2021-10-16T11:17:37,027041-07:00 [   2238]     0  2238    88059      798   155648        0             0 NetworkManager
2021-10-16T11:17:37,027043-07:00 [ 115579]  1000 115579  5533993  1273319 13316096        0             0 lotus-worker
2021-10-16T11:17:37,027044-07:00 [ 115902]  1000 115902  5598286  1442641 14794752        0             0 lotus-worker
2021-10-16T11:17:37,027045-07:00 [ 120439]  1000 120439     4701      482    77824        0             0 systemd
2021-10-16T11:17:37,027047-07:00 [ 120440]  1000 120440    43113     1296   106496        0             0 (sd-pam)
2021-10-16T11:17:37,027048-07:00 [ 120448]  1000 120448   127838     2071   180224        0             0 tracker-miner-f
2021-10-16T11:17:37,027049-07:00 [ 120473]  1000 120473     1848      145    53248        0             0 dbus-daemon
2021-10-16T11:17:37,027050-07:00 [ 120479]  1000 120479    59928      224    98304        0             0 gvfsd
2021-10-16T11:17:37,027051-07:00 [ 120496]  1000 120496    94586      152    98304        0             0 gvfsd-fuse
2021-10-16T11:17:37,027053-07:00 [ 120507]  1000 120507   232105      534   167936        0             0 gvfs-udisks2-vo
2021-10-16T11:17:37,027054-07:00 [ 120530]  1000 120530    79181      250   114688        0             0 gvfs-afc-volume
2021-10-16T11:17:37,027055-07:00 [ 120542]  1000 120542    58957      158    98304        0             0 gvfs-mtp-volume
2021-10-16T11:17:37,027056-07:00 [ 120559]  1000 120559    59526      189    90112        0             0 gvfs-gphoto2-vo
2021-10-16T11:17:37,027057-07:00 [ 120588]  1000 120588    58969      134    90112        0             0 gvfs-goa-volume
2021-10-16T11:17:37,027058-07:00 [ 120592]  1000 120592   137266     1530   258048        0             0 goa-daemon
2021-10-16T11:17:37,027059-07:00 [ 120656]  1000 120656    78689      308   106496        0             0 goa-identity-se
2021-10-16T11:17:37,027061-07:00 [ 121447]   101 121447     6008     1038    81920        0             0 systemd-resolve
2021-10-16T11:17:37,027062-07:00 [ 144994]     0 144994     5848      724    69632        0         -1000 systemd-udevd
2021-10-16T11:17:37,027063-07:00 [ 145257]   102 145257    22565      212    81920        0             0 systemd-timesyn
2021-10-16T11:17:37,027064-07:00 [ 146817]     0 146817  1284318     2814   753664        0          -900 snapd
2021-10-16T11:17:37,027066-07:00 [ 147633]     0 147633     4216      250    69632        0             0 systemd-logind
2021-10-16T11:17:37,027067-07:00 [ 193561]  1000 193561 74131543 54886862 536453120        0             0 lotus-worker
2021-10-16T11:17:37,027069-07:00 [ 197573]     0 197573     3497      367    69632        0             0 sshd
2021-10-16T11:17:37,027070-07:00 [ 197727]  1000 197727     3530      372    69632        0             0 sshd
2021-10-16T11:17:37,027071-07:00 [ 197728]  1000 197728     2722      429    53248        0             0 bash
2021-10-16T11:17:37,027072-07:00 [ 197928]     0 197928     7110      385    81920        0             0 cupsd
2021-10-16T11:17:37,027074-07:00 [ 197929]     0 197929    45111      392   122880        0             0 cups-browsed
2021-10-16T11:17:37,027076-07:00 [ 198786]     0 198786    94816      938   176128        0             0 fwupd
2021-10-16T11:17:37,027077-07:00 oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/system.slice/lotusworker2.service,task=lotus-worker,pid=193561,uid=1000
2021-10-16T11:17:37,027325-07:00 Out of memory: Killed process 193561 (lotus-worker) total-vm:296526172kB, anon-rss:219503836kB, file-rss:35420kB, shmem-rss:8192kB, UID:1000 pgtables:523880kB oom_score_adj:0
2021-10-16T11:17:46,694074-07:00 rfkill: input handler disabled
2021-10-16T11:18:04,521195-07:00 oom_reaper: reaped process 193561 (lotus-worker), now anon-rss:0kB, file-rss:68876kB, shmem-rss:8192kB
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by Reiers** [comment]
Hi @William8Work 

Thanks for the issue ticket. Im unable to reproduce the issue here.
Tried two different 1080ti, with CUDA. No memory leak, or shut down. 

_It can also be that our hw from space race days are starting to its age._ 

```
lotus version
Daemon:  1.17.0+mainnet+git.2830429ad+api1.5.0
Local: lotus version 1.17.0+mainnet+git.2830429ad
```


Closing issue ticket - if you encounter the issue again in current release, please open a new ticket and provide new logs. 
Thanks !
